ca/config.go

```diff
@@ -189,7 +189,7 @@ func GenerateJoinToken(rootCA *RootCA) string {
 
 func getCAHashFromToken(token string) (digest.Digest, error) {
 	split := strings.Split(token, "-")
-	if len(split) != 4 || split[0] != "SWMTKN" || split[1] != "1" {
+	if len(split) != 4 || split[0] != "SWMTKN" || split[1] != "1" || len(split[2]) != base36DigestLen || len(split[3]) != maxGeneratedSecretLength {
 		return "", errors.New("invalid join token")
```

Comment:
I wonder whether that may be too much information to give in an error message, just because the the swarm token is:  `SWMTKN-<version>-<other data>`, and depending on what version the token is, the rest of the token may be different.